### PR TITLE
Add Swift 5.4.2 toolchain

### PR DESCRIPTION
**Pull request by compnerd**
This introduces a winget package for the Swift toolchain.  The
dependencies are:
- Windows SDK (10.0.17763.0+)
- BuildTools (for VC++ SDK)
- Git
- Python[=3.7.8] (required for the debugger)

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/19313)

### CODE REVIEW

File: manifests/s/Swift/Toolchain/5.4.2/Swift.Toolchain.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿# yaml-language-server: $schema=https://aka.ms/winget-manifest.version.1.0.0.schema.json
+PackageIdentifier: Swift.Toolchain
+PackageVersion: 5.4.2
+PackageName: Swift Toolchain
+PackageLocale: en-US
+Author: Swift Open Source Project
+Publisher: Swift Open Source Project
```
Comment (OfficialEsco) [inline manifests/s/Swift/Toolchain/5.4.2/Swift.Toolchain.yaml:7]:
```suggestion
Publisher: dt.compnerd.org
```

**Review by OfficialEsco — CHANGES_REQUESTED** [review]
image

edit: oh hey you're the developer, you should change the Publisher and Version which shows up in Windows control panel.
`PackageName`, `Publisher` and `Version` in the Manifest needs to match what Windows sees for `winget list` and `winget upgrade` to work.

**Comment by compnerd** [comment]
@OfficialEsco thanks for the suggestion, I've applied that.  For now, I haven't worked out the versioning as the MSI versioning scheme is pretty limiting and doesn't match what Swift actually uses.  Currently, the MSI package always returns the same value, though the package itself is versioned differently.  Is there a good way to handle this?

**Review by OfficialEsco — APPROVED** [review]
Can't help with any Software Development questions 😁 Thought the Developers of the software knew their way around, but i keep seeing software not reporting correct version/version at all or Publisher being wrong so maybe its something you setup once very early and forget 🤷‍♂️

**Pull request merged**